src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+                        (readGroupID.startsWith(laneID + ".") ||   // lane is the same, but sample identifier is different
+                                readGroupID.equals(laneID)))               // in case there is no sample identifier, they have to be exactly the same
+                {
+                    filteredTracker.add(p);
+                }
+            } else {
+                if (readGroupID == null) {
+                    filteredTracker.add(p);
+                }
+            }
+        }
+        return filteredTracker.size() > 0 ? new ReadPileup(loc, filteredTracker) : null;
+    }
+
+    /**
+     * Gets a set of the read groups represented in this pileup.
+     * Note: contains null if a read has a null read group.
+     */
+    public Set<String> getReadGroupIDs() {
+        return pileupElements.stream().map(pe -> pe.getRead().getReadGroup()).collect(Collectors.toSet());
+    }
+
+    /**
+     * Gets a set of the samples represented in this pileup.
+     * Note: contains null if a read has a null read group or a null sample name.
+     */
+    public Set<String> getSamples(final SAMFileHeader header) {
+        return pileupElements.stream().map(pe -> pe.getRead()).map(r -> ReadUtils.getSampleName(r, header)).collect(Collectors.toSet());
+    }
+
+
+    /**
+     * The best way to access PileupElements where you only care about the bases and quals in the pileup.
+     * <p/>
+     * for (PileupElement p : this) { doSomething(p); }
+     * <p/>
+     * Provides efficient iteration of the data.
+     */
+    public Iterator<PileupElement> iterator() {
+        return Collections.unmodifiableList(pileupElements).iterator();
+   }
+
+    /**
+     * The number of elements in this pileup.
+     */
+    public int size() {
+        return pileupElements.size();
+    }
+
+    /**
+     * @return true if there are 0 elements in the pileup, false otherwise
+     */
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    /**
+     * @return the location of this pileup
+     */
+    public Locatable getLocation() {
+        return loc;
+    }
+
+    /**
+     * Get counts of A, C, G, T in order, which returns a int[4] vector with counts according
+     * to BaseUtils.simpleBaseToBaseIndex for each base.
+     */
+    public int[] getBaseCounts() {
+        final int[] counts = new int[4];
+
+        for (final PileupElement pile : this) {
+            // skip deletion sites
+            if (!pile.isDeletion()) {
+                final int index = BaseUtils.simpleBaseToBaseIndex(pile.getBase());
+                if (index != -1) {
+                    counts[index]++;
+                }
+            }
+        }
+
+        return counts;
+    }
+
+    /**
+     * Creates a pileup formar string.
+     * In the pileup format, each line represents a genomic position, consisting of chromosome name,
+     * coordinate, reference base, read bases, read qualities and alignment mapping qualities.
```

Comment:
This method appears to be tied to an obsolete file format? Do we really need it? Can we just implement a standard `toString()` instead?